File: python/46_goldbachs_other_conjecture.py
from math import sqrt

def goldbachs_other_conjecture() -> int:

    counterexample: int = 0
    n: int = 35
    
    while counterexample == 0:
        # check composite odds only
        if is_prime(n):
            n += 2
            continue
        
        found: bool = False
        s: int = 1
        
        # n = p + 2*s*s
        # p = n - 2*s*s
        
        # test differences of n and all doubled squares less than n
        while (2*s*s < n):
            if is_prime(n - 2*s*s):
                found = True
                break
            s += 1
        
        if not found:
            counterexample = n
        n += 2
    
    return counterexample
    
def is_prime(n: int) -> bool:
    if n < 2:
        return False
    
    i: int = 2
    while i*i <= n:
        if n % i == 0:
            return False
        i += 1
    return True

if __name__ == "__main__":
    print(goldbachs_other_conjecture())

File: c#/_46_GoldbachsOtherConjecture.cs
using System;

public class _46_GoldbachsOtherConjecture{
	public static int Euler46(){
		// n = p + 2x^2, x^2 = s
		// p = n - 2s
		int n = 9, res = 0;
		while (res == 0){
			if (!IsPrime(n)){
				bool found = false;
				for (int i = 1; 2 * i * i < n; i++){
					int s = i * i;
					if (IsPrime(n - 2 * s)){
						found = true;
						break;
					}
				}
				if (!found) res = n;
			}
			n += 2;
		}
		return res;
	}
	
	private static bool IsPrime(int n){
		for (int i = 2; i * i <= n; i++){
			if (n % i == 0) return false;	
		}
		return true;
	}
	
	public static void Main(){
		Console.WriteLine(Euler46());
	}
}
